STYLE.md

```diff
@@ -0,0 +1,453 @@
+Haskell Style Guide
+============================
+> This style guide is mostly a copy of [Johan Tibell's guide](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md)
+> with some restructurization, elaboration on some topics and some additions.
+> This style guide's aims are code beauty, readability and understandability.
+
+
+Principles
+----------
+* The primary goal is readability/maintainability with performance as a close second.
+* Always use types to clarify intent and make invalid states unrepresentable.
+* Use type signatures as compiler checked documentation.
+* Break up large functions into smaller sub functions.
+* When modifying or adding code to existing code (either at the file, project or company level), try to match the style of the existing code.
+
+General guide lines
+-------------------
+### Line Length
+
+Maximum line length is *80 characters* or *100 characters* if necessary.
+
+Modern screens have high definition and big width.
+But with some tiling managers and two terminals on one screen you are not able to
+see many characters on one line.
+On the other hand, restricting line size to a very small number like 80 leads to
+some crazy indentation despite the fact that
+shorter lines should force you to write well structured code.
+That's why *100* is a reasonable compromise.
+
+
+### Indentation
+
+* Do not vertically align code. Mainly because the need to change identifiers and re-aligning the other lines causes semantically null changes in the diff.
+* Tabs are illegal.
+* Use spaces for indenting.
+* Indent your code blocks with *2 spaces*.
+* Indent the `where` keyword with one space to set it
+apart from the rest of the code and indent the definitions in a
+`where` clause with 1 space.
+
+Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+  name <- getLine
+  putStrLn $ greeting name
+ where
+  greeting :: String -> String
+  greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ [] = []
+filter p (x:xs)
+  | p x = x : filter p xs
+  | otherwise = filter p xs
+```
+
+### Blank Lines
+
+* One blank line between top-level definitions.
+* No blank lines between type signatures and function definitions.
+* Add one blank line between
+functions in a type class instance declaration if the function bodies
+are large.
+* You can add blank lines inside a big `do` block to separate logical
+parts of it.
+* You can also use blank lines to separate definitions inside `where`
```

Comment:
@Jimbo4350, this conflicts with https://github.com/input-output-hk/cardano-node/pull/164/files#diff-32430379d7ba797ddf827c29a4eac4c1R38, which mandates the blank line separation.

Reply:
@erikd, what is your opinion on the 1-blank-line mandatory separation of where clauses?

Reply:
I think hard rules don't make a lot of sense. In some cases a single blank line does make sense (eg to separate two 10 line functions, with type signatures) and in other cases it doesn't (eg 3 single line clauses without type sigs).

Sorry for not giving a straight yes/not answer :)

Reply:
I'll remove it :+1: